polaris-react/src/components/Checkbox/Checkbox.scss

```diff
@@ -50,7 +50,10 @@
       // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
       @include control-backdrop(disabled);
       #{$se23} & {
-        border: none;
+        // We set the border-color to transparent here
+        // in order for the background color to bleed all the way to the edge of the element.
+        border-color: transparent;
```

Comment:
@qt314  just fyi I've removed the `border:none` line on disabled checkboxes in favor of `border:transparent` 
to ensure the fill is consistent across `checked` and `unchecked` states. 

Before: 
![Screenshot 2023-07-06 at 10 57 36 am](https://github.com/Shopify/polaris/assets/12119389/e019a325-34b3-41ad-88ae-579a319a846e)

After: 
![Screenshot 2023-07-06 at 10 57 52 am](https://github.com/Shopify/polaris/assets/12119389/0d9d12ff-715b-4709-81c3-6dd40c7404a2)